### ISSUE TITLE
Getting parser error

### ISSUE THREAD

**Issue by anayyar82** [SimonJPegg/XMLtoAvroSpark#1]
Caused by: org.xml.sax.SAXParseException; lineNumber: 85; columnNumber: 16; Invalid byte 2 of 4-byte UTF-8 sequence.

**Comment by SimonJPegg** [comment]
Can you provide an example of the XML?

**Comment by anayyar82** [comment]
Hello Simon, I am able to parse the large xml and generate the desire output. 

Now another issue is coming :   Any idea how to tackle this ... 

java.lang.OutOfMemoryError
	at java.io.ByteArrayOutputStream.hugeCapacity(ByteArrayOutputStream.java:123)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:117)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.spark-project.guava.io.ByteStreams.copy(ByteStreams.java:211)
	at org.spark-project.guava.io.ByteStreams.toByteArray(ByteStreams.java:252)
	at org.apache.spark.input.WholeTextFileRecordReader.nextKeyValue(WholeTextFileRecordReader.scala:83)
	at org.apache.hadoop.mapreduce.lib.input.CombineFileRecordReader.nextKeyValue(CombineFileRecordReader.java:69)
	at org.apache.spark.rdd.NewHadoopRDD$$anon$1.hasNext(NewHadoopRDD.scala:163)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply$mcV$sp(PairRDDFunctions.scala:1034)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply(PairRDDFunctions.scala:1034)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12$$anonfun$apply$4.apply(PairRDDFunctions.scala:1034)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1206)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12.apply(PairRDDFunctions.scala:1042)
	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$saveAsNewAPIHadoopDataset$1$$anonfun$12.apply(PairRDDFunctions.scala:1014)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
	at org.apache.spark.scheduler.Task.run(Task.scala:88)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

**Comment by SimonJPegg** [comment]
Looks like the executor is running out of ram.  Try playing with the

spark.executor.memory
spark.driver.memory

properties, you can set these either through the SparkConf or the spark-submit command.

**Comment by anayyar82** [comment]
sure will try to add. 

One more question :  Can i change avro to parquet ? 

e.g. below, some how it didnt work , any idea ?

 ParquetOutputFormat.setWriteSupportClass(job, classOf[AvroWriteSupport])
      AvroParquetOutputFormat.setSchema(job, protocol.getType("Element"))

    parsedXMLRDD.map(elem => {
        (null,elem)
      }).saveAsNewAPIHadoopFile(parquetFileLocation,
        classOf[Void],
        classOf[GenericData.Record],
        classOf[ParquetOutputFormat[GenericData.Record]],
        job.getConfiguration)
      0
    } catch {
      case ex : Throwable => {
        ex.printStackTrace
        -1
      }
    }

**Comment by SimonJPegg** [comment]
If you'd like to transform XML to Parquet have a look at SparkSQL and the Spark-XML library from Databricks 

it's as simple as 
val sc = new SparkContext
val sqlContext = new SQLContext(sc)
val df = sqlContext.read.format("com.databricks.spark.xml").option("rowTag", "book").load("books.xml")
df.write("/some/path") 

the default output format is parquet.

**Comment by anayyar82** [comment]
yes but here the advantage is we don't need to take care of xml, dynamically if xml tags added or deleted can be loaded with new schema. 

so if we can address with this code with parquet format will be great help....  

issue with databricks library they are not working well with nested structure.
